### PR TITLE
[MRG] fixed bugs in preproc.py

### PR DESCRIPTION
**Pull request by annapasca**
- added ref_meg='auto'
- insert try for plot_psd

### CODE REVIEW

File: ephypype/preproc.py [hunk 1]
```diff
@@ -427,18 +430,21 @@ def generate_report(raw, ica, subj_name, basename,
...
+        for i_ic in ic_nums:
+            fig = ica_src.plot_psd(tmax=60, picks=[i_ic], fmax=140, show=False)
+            fig.set_figheight(3)
+            fig.set_figwidth(5)
+            psds.append(fig)
+            captions_psd.append('IC #' + str(i_ic))
+
+        report.add_figs_to_section(figs=psds, captions=captions_psd,
+                                   section='ICA - muscles')
+    except:  # noqa
```
Comment (jasmainak) [inline ephypype/preproc.py]: it's not a good idea to put `try-except` clause in your repo. It's better to do it with an `if-else` ... and raise an appropriate warning

Comment (annapasca) [reply ephypype/preproc.py]: plot_psd raises an error. Looking at the MNE code it seems due to the fact that now to plot the PSD of 1 ICA component we have to use directly psd_multipaper. @dmalt do you have time to ckeck this? 
@jasmainak the try was a temporary fix :)

**Comment by annapasca** [comment]
@dmalt @jasmainak I removed the try and substitute the plot_psd with psd_multipaper + plot

**Comment by annapasca** [comment]
I added some unit test...could they be ok?

**Comment by annapasca** [comment]
@jasmainak it seems we need scikit-learn

File: ephypype/preproc.py [hunk 2]
```diff
@@ -427,17 +432,29 @@ def generate_report(raw, ica, subj_name, basename,
...
     captions_psd = []
     ica_src = ica.get_sources(raw)
     for i_ic in ic_nums:
-        fig = ica_src.plot_psd(tmax=60, picks=[i_ic], fmax=140, show=False)
-        fig.set_figheight(3)
-        fig.set_figwidth(5)
-        psds.append(fig)
+        # fig = ica_src.plot_psd(tmax=60, picks=[i_ic], fmax=140, show=False)
+        # fig.set_figheight(3)
+        # fig.set_figwidth(5)
```
Comment (jasmainak) [inline ephypype/preproc.py]: remove these comments when you are ready to merge

Comment (jasmainak) [reply ephypype/preproc.py]: and set PR to [MRG] when you are ready :)

**Comment by jasmainak** [comment]
@annapasca you will need to update `.travis.yml` and `appveyor.yml`. Feel free to update them so that `scikit-learn` is installed.

**Comment by jasmainak** [comment]
Thanks @annapasca for adding tests. That is awesome !! :tada:

**Comment by annapasca** [comment]
the error in appveyor it seems due to some download problem of MNE, is it? How can I start again appveyor?

**Comment by jasmainak** [comment]
@annapasca you may want to add a tiny commit or amend the last commit and force push. I don't think there is a way to restart the build like in Travis ...

**Comment by annapasca** [comment]
@jasmainak can I restart travis? It says the build has terminated but no error specified

**Comment by jasmainak** [comment]
@annapasca sure feel free to restart Travis if you need. But this error is specific. It says:

> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Your tests should ideally be fast so that the screen is updating all the time. 10 minutes is the upper limit ...

**Comment by annapasca** [comment]
> Your tests should ideally be fast so that the screen is updating all the time. 10 minutes is the upper limit ...

@jasmainak thks! I understood now! I tried to make the test faster

**Comment by jasmainak** [comment]
also @annapasca try to give your feature branch a more meaningful name if you can :-) Otherwise it's hard to try two pull requests which have the same name as the feature branch ...

**Comment by jasmainak** [comment]
Few more comments:

- tests should not write to the directory we are working from. Ideally it should save stuff in 'temp' directory
- tests should not throw any warnings. Right now, I get warnings of too many open windows

But I'd say not urgent for now. Let's raise an issue so we don't forget.

**Comment by annapasca** [comment]
@jasmainak about test

> tests should not write to the directory we are working from. Ideally it should save stuff in 'temp' directory

NeuroPycon works saving most output in the absolute path, thus how can we solve this issue for the test?
e.g. the `test_preproc` saves 4 files in the dir we are running the test...

> tests should not throw any warnings. Right now, I get warnings of too many open windows

OK! How is possible to avoid this?

Thks a lot for all your useful suggestions! :)

**Pull request merged**